website/common/script/ops/buyGear.js

```diff
@@ -35,6 +35,15 @@ module.exports = function buyGear (user, req = {}, analytics) {
     throw new NotAuthorized(i18n.t('equipmentAlreadyOwned', req.language));
   }
 
+  let itemIndex = item.index;
```

Comment:
This new check seems right but I think there should be a check for the cases where item.index is not a number, like here 
<img width="683" alt="schermata 2017-11-27 alle 13 01 46" src="https://user-images.githubusercontent.com/589911/33265762-27768ad4-d373-11e7-8a3b-b893ed292755.png">

Maybe we could simply exclude from the check items when `Number.isNaN(Number(item.index)) === true` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN)

Reply:
I tried both ways to check for NaN and I keep getting that `isNan()` is not a function. Since NaN is the only object that doesn't equal itself, I checked for NaN this way.

Let me know what you think.